code/game/objects/items/food/bread.dm

```diff
@@ -0,0 +1,309 @@
+	AddComponent(/datum/component/edible,\
+			initial_reagents = list_reagents,\
+			food_flags = food_flags,\
+			foodtypes = foodtypes,\
+			volume = volume,\
+			eat_time = eat_time,\
+			tastes = tastes,\
+			eatverbs = eatverbs,\
+			bite_consumption = bite_consumption,\
+			on_consume = CALLBACK(src, .proc/On_Consume))
+
+
+/obj/item/food/deepfryholder/Initialize(mapload, obj/item/fried)
+	. = ..()
+	name = fried.name //We'll determine the other stuff when it's actually removed
+	appearance = fried.appearance
+	layer = initial(layer)
+	plane = initial(plane)
+	lefthand_file = fried.lefthand_file
+	righthand_file = fried.righthand_file
+	inhand_icon_state = fried.inhand_icon_state
+	desc = fried.desc
+	w_class = fried.w_class
+	slowdown = fried.slowdown
+	equip_delay_self = fried.equip_delay_self
+	equip_delay_other = fried.equip_delay_other
+	strip_delay = fried.strip_delay
+	species_exception = fried.species_exception
+	item_flags = fried.item_flags
+	obj_flags = fried.obj_flags
+	inhand_x_dimension = fried.inhand_x_dimension
+	inhand_y_dimension = fried.inhand_y_dimension
+
+	if(!(SEND_SIGNAL(fried, COMSIG_ITEM_FRIED, src) & COMSIG_FRYING_HANDLED)) //If frying is handled by signal don't do the defaault behavior.
+		fried.forceMove(src)
+
+
+/obj/item/food/deepfryholder/Destroy()
+	if(contents)
+		QDEL_LIST(contents)
+	. = ..()
+
+/obj/item/food/deepfryholder/proc/On_Consume(eater, feeder)
+	if(contents)
+		QDEL_LIST(contents)
+	..()
+
+/obj/item/food/deepfryholder/proc/fry(cook_time = 30)
+	switch(cook_time)
+		if(0 to 15)
+			add_atom_colour(rgb(166,103,54), FIXED_COLOUR_PRIORITY)
+			name = "lightly-fried [name]"
+			desc = "[desc] It's been lightly fried in a deep fryer."
+		if(16 to 49)
+			add_atom_colour(rgb(103,63,24), FIXED_COLOUR_PRIORITY)
+			name = "fried [name]"
+			desc = "[desc] It's been fried, increasing its tastiness value by [rand(1, 75)]%."
+		if(50 to 59)
+			add_atom_colour(rgb(63,23,4), FIXED_COLOUR_PRIORITY)
+			name = "deep-fried [name]"
+			desc = "[desc] Deep-fried to perfection."
+		if(60 to INFINITY)
+			add_atom_colour(rgb(33,19,9), FIXED_COLOUR_PRIORITY)
+			name = "\proper the physical manifestation of the very concept of fried foods"
+			desc = "A heavily-fried...something.  Who can tell anymore?"
+	foodtypes |= FRIED
+
+/obj/item/food/butterbiscuit
+	name = "butter biscuit"
+	desc = "Well butter my biscuit!"
+	icon = 'icons/obj/food/food.dmi'
+	icon_state = "butterbiscuit"
+	list_reagents = list(/datum/reagent/consumable/nutriment = 6, /datum/reagent/consumable/nutriment/vitamin = 1)
+	tastes = list("butter" = 1, "biscuit" = 1)
+	foodtypes = GRAIN | BREAKFAST
+
+/obj/item/food/butterdog
+	name = "butterdog"
+	desc = "Made from exotic butters."
+	icon = 'icons/obj/food/food.dmi'
+	icon_state = "butterdog"
+	bite_consumption = 1
+	list_reagents = list(/datum/reagent/consumable/nutriment = 6, /datum/reagent/consumable/nutriment/vitamin = 1)
+	tastes = list("butter", "exotic butter")
```

Comment:
Do these require an assigned value?

Reply:
??

Reply:
Compare with `tastes = list("butter" = 1, "biscuit" = 1)` in `/obj/item/food/butterbiscuit`

Reply:
guess so